### PR TITLE
(BKR-70) added error condition handling to SshConnection.close

### PR DESCRIPTION
**Pull request by kevpl**

### CODE REVIEW

File: lib/beaker/ssh_connection.rb [hunk 1]
```diff
@@ -62,10 +62,21 @@ def connect
     def close
       begin
         @ssh.close if @ssh
-      rescue
+      rescue IOError => detail
+        if detail.message == 'closed stream'
+          @logger.warn 'Attemped ssh.close.  Caught IOError.close stream.  Destroying ssh instance.'
+        else
+          @logger.warn 'ssh.close threw unexpected IOError.  Shutting down, and re-raising error below'
```
Comment (anodelman) [inline lib/beaker/ssh_connection.rb]: Add detail message in here.

File: lib/beaker/ssh_connection.rb [hunk 2]
```diff
@@ -62,10 +62,21 @@ def connect
...
+          @ssh.shutdown!
+          raise detail
+        end
+      rescue => detail
+        @logger.warn 'ssh.close threw unexpected Error.  Shutting down, and re-raising error below'
         @ssh.shutdown!
```
Comment (anodelman) [inline lib/beaker/ssh_connection.rb]: Won't this attempt shutdown twice?

File: lib/beaker/ssh_connection.rb (hunk 1 above)
Comment (kevpl) [inline lib/beaker/ssh_connection.rb]: sure

File: lib/beaker/ssh_connection.rb (hunk 2 above)
Comment (kevpl) [inline lib/beaker/ssh_connection.rb]: no, execution shouldn't enter multiple rescue blocks

File: lib/beaker/ssh_connection.rb [hunk 3]
```diff
@@ -62,10 +62,21 @@ def connect
...
+      rescue IOError => detail
+        if detail.message == 'closed stream'
+          @logger.warn 'Attemped ssh.close.  Caught IOError.close stream.  Destroying ssh instance.'
+        else
+          @logger.warn "ssh.close threw unexpected IOError: #{detail.message}.  Shutting down, and re-raising error below"
+          @ssh.shutdown!
+          raise detail
+        end
+      rescue => detail
+        @logger.warn 'ssh.close threw unexpected Error.  Shutting down, and re-raising error below'
```
Comment (anodelman) [inline lib/beaker/ssh_connection.rb]: detail.message here as well.

**Comment by anodelman** [comment]
Tested locally and doesn't handle a SUT rebooting during a test run.  Will open a new PR with updated code.

**Pull request closed without merging**